pkg/api/handlers/compat/images.go

```diff
@@ -356,6 +356,14 @@ loop: // break out of for/select infinite loop
 					Message: msg,
 				}
 				report.ErrorMessage = msg
+				var errcd errcode.ErrorCoder
+				var statusCode int
+				if errors.As(err, &errcd) {
+					statusCode = errcd.ErrorCode().Descriptor().HTTPStatusCode
```

Comment:
https://github.com/containers/podman/pull/16097#issuecomment-1276727298

Sure, this is better than nothing.